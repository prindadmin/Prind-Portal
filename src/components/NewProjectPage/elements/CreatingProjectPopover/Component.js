import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
