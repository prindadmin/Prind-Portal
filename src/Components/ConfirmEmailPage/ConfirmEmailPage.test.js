
import { shallow } from 'enzyme';
import Component from './ConfirmEmailPage'
import ReactGA from 'react-ga';

var props;

beforeEach(() => {
  ReactGA.initialize('dummy', { testMode: true });
  props = {
    confirmUser: jest.fn(),
    location: {
      search: "?user_name=1234567890&confirmation_code=abcdef",
      pathname: '/confirm-email'
    },
    history: {
      push: jest.fn()
    }
  };
});

it('ConfirmEmailPage - Should render', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ConfirmEmailPage - Should render but no search terms were provided', () => {
  props.location.search = ""
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ConfirmEmailPage - Should confirm user', () => {
  const mockConfirmUser = jest.fn((parameters, resolve, reject) => {
    resolve()
  })
  props.confirmUser = mockConfirmUser
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});

it('ConfirmEmailPage - Should reject confirmation of user', () => {
  const mockConfirmUser = jest.fn((parameters, resolve, reject) => {
    reject()
  })
  props.confirmUser = mockConfirmUser
  const component = shallow(<Component {...props} />);
  expect(mockConfirmUser).toHaveBeenCalledWith({ user_name: "1234567890", confirmation_code: "abcdef" }, expect.any(Function), expect.any(Function))
  expect(mockConfirmUser).toHaveBeenCalledTimes(1)
  expect(component).toMatchSnapshot();
});

it('ConfirmEmailPage - Should reject confirmation of user then retry', () => {
  const mockConfirmUser = jest.fn((parameters, resolve, reject) => {
    reject()
  })
  props.confirmUser = mockConfirmUser
  const component = shallow(<Component {...props} />);
  expect(mockConfirmUser).toHaveBeenCalledWith({ user_name: "1234567890", confirmation_code: "abcdef" }, expect.any(Function), expect.any(Function))
  expect(mockConfirmUser).toHaveBeenCalledTimes(1)
  const retryButton = component.find('#retry-button')
  retryButton.simulate('click')
  expect(mockConfirmUser).toHaveBeenCalledTimes(2)
  expect(component).toMatchSnapshot();
});
