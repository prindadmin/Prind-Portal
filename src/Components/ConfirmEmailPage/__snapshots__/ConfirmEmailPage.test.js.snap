// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmEmailPage - Should confirm user 1`] = `
<div
  id="confirming-email-address"
>
  <ProjectLoading
    text="Confirming your email address, please wait..."
  />
</div>
`;

exports[`ConfirmEmailPage - Should reject confirmation of user 1`] = `
<div
  id="confirming-email-address"
>
  <div
    className="error-container"
  >
    <Blueprint3.Callout
      intent="danger"
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <div>
        There was an error confirming your email address.  Please check that the link has not expired and try again.
      </div>
    </Blueprint3.Callout>
    <Blueprint3.ButtonGroup
      fill={true}
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <Blueprint3.Button
        id="retry-button"
        intent="primary"
        onClick={[Function]}
        text="Retry"
      />
    </Blueprint3.ButtonGroup>
  </div>
</div>
`;

exports[`ConfirmEmailPage - Should reject confirmation of user then retry 1`] = `
<div
  id="confirming-email-address"
>
  <div
    className="error-container"
  >
    <Blueprint3.Callout
      intent="danger"
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <div>
        There was an error confirming your email address.  Please check that the link has not expired and try again.
      </div>
    </Blueprint3.Callout>
    <Blueprint3.ButtonGroup
      fill={true}
      style={
        Object {
          "marginBottom": "15px",
        }
      }
    >
      <Blueprint3.Button
        id="retry-button"
        intent="primary"
        onClick={[Function]}
        text="Retry"
      />
    </Blueprint3.ButtonGroup>
  </div>
</div>
`;

exports[`ConfirmEmailPage - Should render 1`] = `
<div
  id="confirming-email-address"
>
  <ProjectLoading
    text="Confirming your email address, please wait..."
  />
</div>
`;

exports[`ConfirmEmailPage - Should render but no search terms were provided 1`] = `
<div
  id="confirming-email-address"
>
  <ProjectLoading
    text="Confirming your email address, please wait..."
  />
</div>
`;
